Make test_report_with_reconnected_reporter_but_finished_jobs optional to Mac

**Describe the bug**
Flakiness was observed when running `test_report_with_reconnected_reporter_but_finished_jobs` in Mac

> tests/unit_tests/c_wrappers/job_runner/test_event_reporter.py::test_report_with_reconnected_reporter_but_finished_jobs - AssertionError: 10 seconds should be sufficient to send three events
> assert (1669881258.64687 + 10) > 1669881268.756746
>  where 1669881268.756746 = <built-in function time>()
>   where <built-in function time> = time.time
> = 1 failed, 1089 passed, 53 skipped, 4 xfailed, 70 warnings in 442.96s (0:07:22) =
> Error: Process completed with exit code 1.

**Fix**
Skip this test if it is macOS, we evaluate it was a performance issue.

--- a/tests/unit_tests/c_wrappers/job_runner/test_event_reporter.py
+++ b/tests/unit_tests/c_wrappers/job_runner/test_event_reporter.py
@@ -1,5 +1,6 @@
 import json
 import os
+import sys
 import time
 from unittest.mock import patch
 
@@ -199,6 +200,9 @@ def test_report_with_failed_reporter_but_finished_jobs(unused_tcp_port):
     assert len(lines) == 0, "expected 0 Job running messages"
 
 
+@pytest.mark.skipif(
+    sys.platform.startswith("darwin"), reason="Performance can be flaky"
+)
 def test_report_with_reconnected_reporter_but_finished_jobs(unused_tcp_port):
     # this is to show when the reporter fails but reconnects
     # reporter still manages to send events and completes fine

